test/common/common.go

```diff
@@ -0,0 +1,71 @@
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	. "github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/gomega"
+)
+
+// RunGetDevices executes "get devices" CLI command with optional arguments.
+// Verifies if the expected error condition is met and checks if the output contains a given substring.
+func RunGetDevices(harness *e2e.Harness, expectError bool, expectedSubstring string, args ...string) (string, error) {
+	baseArgs := []string{"get", "devices"}
+	args = append(baseArgs, args...)
+	out, err := harness.CLI(args...)
+
+	if expectError {
+		Expect(err).To(HaveOccurred())
```

Comment:
Please, don't call _Expect()_ from the util functions. Calling Expect() from the helper function is anti-pattern and can cause flaky behavior, panic, or produce unhelpful test output.